streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -70,6 +70,10 @@ public StreamsMetadataState(final TopologyBuilder builder) {
     public synchronized Collection<StreamsMetadata> getAllMetadataForStore(final String storeName) {
         Objects.requireNonNull(storeName, "storeName cannot be null");
 
+        if (!isInitialized()) {
+            return Collections.emptyList();
```

Comment:
@guozhangwang i wasn't sure what to do in this case, but if we decided to not throw exceptions in this class then an empty list seems valid.